py/ringbuf.c

```diff
@@ -71,3 +71,49 @@ int ringbuf_put16(ringbuf_t *r, uint16_t v) {
     r->iput = iput_b;
     return 0;
 }
+
+#include <string.h>
+
+// Returns:
+//    1: Success
+//    0: Not enough data available to complete read (try again later)
+//   -1: Requested read is larger than buffer - will never succeed
+int ringbuf_read(ringbuf_t *r, void *data, size_t data_len) {
```

Comment:
Need to think if these should be named like `ringbuf_get_bytes`/`ringbuf_put_bytes`, and instead of `void *data` take a `uint8_t *data`.  Or at least called `ringbuf_get_data`/`ringbuf_put_data`, to match the existing functions.

Also IMO it should return 0 to indicate success, like `ringbuf_put16`.

Reply:
Changed to `ringbuf_get_bytes()/ringbuf_put_bytes()` and use `uint8_t *data`.
Changed return value to 0 on success.